### PR TITLE
Bug 1463852 - List images required to install Openshift disconnected environment 

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1463852

### CODE REVIEW

**Comment by mburke5678** [comment]
@jianlinliu @miheer Can you help with this BZ?
https://bugzilla.redhat.com/show_bug.cgi?id=1463852

**Comment by mburke5678** [comment]
@miheer PTAL

**Comment by mburke5678** [comment]
@miheer @jianlinliu PTAL. 

Miheer asked about S2I and xpass images. We have a step that lists S2I and JBoss images. It that what you are looking for? 

Also, maybe this is a bad question, after downloading all the images, the user needs to add ALL of the images into tarballs?

File: install_config/install/disconnected_install.adoc [hunk 1]
```diff
@@ -320,6 +359,42 @@ and then transporting them:
...
+    registry.access.redhat.com/openshift3/logging-fluentd \
+    registry.access.redhat.com/openshift3/logging-kibana \
+    registry.access.redhat.com/openshift3/metrics-cassandra \
+    registry.access.redhat.com/openshift3/metrics-deployer \
+    registry.access.redhat.com/openshift3/metrics-hawkular-metrics \
+    registry.access.redhat.com/openshift3/metrics-hawkular-openshift-agent \
+    registry.access.redhat.com/openshift3/metrics-heapster \
+    registry.access.redhat.com/openshift3/node \
+    registry.access.redhat.com/openshift3/openvswitch
+----
```
Comment (sferich888) [inline install_config/install/disconnected_install.adoc]: Do we really want to provide a full list like this? Would it not be better to pull the list with a set of AIP calls? 

https://mojo.redhat.com/docs/DOC-1141677

File: install_config/install/disconnected_install.adoc [hunk 2]
```diff
@@ -194,16 +200,26 @@ Replace `<tag>` with `{latest-int-tag}` for the latest version.
 endif::[]
 +
 ----
+# docker pull registry.access.redhat.com/openshift3/container-engine:<tag>
+# docker pull registry.access.redhat.com/openshift3/efs-provisioner:<tag>
+# docker pull registry.access.redhat.com/openshift3/jenkins-1-rhel7:<tag>
```
Comment (jianlinliu) [inline install_config/install/disconnected_install.adoc]: The following jenkins-* images should belong to s2i builder images, but not "components for the additional centralized log aggregation and metrics aggregation components"

File: install_config/install/disconnected_install.adoc [hunk 3]
```diff
@@ -194,16 +200,26 @@ Replace `<tag>` with `{latest-int-tag}` for the latest version.
...
-# docker pull registry.access.redhat.com/openshift3/logging-curator:<tag>
-# docker pull registry.access.redhat.com/openshift3/logging-auth-proxy:<tag>
+# docker pull registry.access.redhat.com/openshift3/logging-kibana:<tag>
+# docker pull registry.access.redhat.com/openshift3/metrics-cassandra:<tag>
 # docker pull registry.access.redhat.com/openshift3/metrics-deployer:<tag>
 # docker pull registry.access.redhat.com/openshift3/metrics-hawkular-metrics:<tag>
-# docker pull registry.access.redhat.com/openshift3/metrics-cassandra:<tag>
+# docker pull registry.access.redhat.com/openshift3/metrics-hawkular-openshift-agent:<tag>
 # docker pull registry.access.redhat.com/openshift3/metrics-heapster:<tag>
+# docker pull registry.access.redhat.com/openshift3/node:<tag>
```
Comment (jianlinliu) [inline install_config/install/disconnected_install.adoc]: node and openvswitch images should belong to "the required {product-title} containerized components"

File: install_config/install/disconnected_install.adoc [hunk 4]
```diff
@@ -194,16 +200,26 @@ Replace `<tag>` with `{latest-int-tag}` for the latest version.
 endif::[]
 +
 ----
+# docker pull registry.access.redhat.com/openshift3/container-engine:<tag>
+# docker pull registry.access.redhat.com/openshift3/efs-provisioner:<tag>
```
Comment (jianlinliu) [inline install_config/install/disconnected_install.adoc]: container-engine and efs-provisioner images should belong to "the required {product-title} containerized components"

File: install_config/install/disconnected_install.adoc [hunk 5]
```diff
@@ -320,6 +359,42 @@ and then transporting them:
...
+    registry.access.redhat.com/openshift3/metrics-cassandra \
+    registry.access.redhat.com/openshift3/metrics-deployer \
+    registry.access.redhat.com/openshift3/metrics-hawkular-metrics \
+    registry.access.redhat.com/openshift3/metrics-hawkular-openshift-agent \
+    registry.access.redhat.com/openshift3/metrics-heapster \
+    registry.access.redhat.com/openshift3/node \
+    registry.access.redhat.com/openshift3/openvswitch
+----
+
+*Is this needed?*
```
Comment (jianlinliu) [inline install_config/install/disconnected_install.adoc]: ASB depends on product registry, so in a disconnected environment, from my understanding, no need to mention this part.

File: install_config/install/disconnected_install.adoc [hunk 6]
```diff
@@ -320,6 +359,42 @@ and then transporting them:
     registry.access.redhat.com/openshift3/metrics-cassandra \
     registry.access.redhat.com/openshift3/metrics-heapster
 ----
++
+----
+# docker save -o ose3-logging-metrics-images.tar \
+    registry.access.redhat.com/openshift3/container-engine \
+    registry.access.redhat.com/openshift3/efs-provisioner \
```
Comment (jianlinliu) [inline install_config/install/disconnected_install.adoc]: just like what I mentioned in my previous comments, some images are put into wrong category.

**Comment by mburke5678** [comment]
@jianlinliu Can you take another look to make sure I got your comments correct? 
The original Bugzilla was filed against OCP 3.5; do the change apply for 3.6 and forward?

**Comment by jianlinliu** [comment]
The changes look good to me, but the image list I provided in the original bug is targeted for 3.6, but not 3.5, so I would say LGTM for 3.6.

**Comment by mburke5678** [comment]
@jianlinliu @sferich888 Can we get a list of images for 3.5 or 3.4? is it worth the effort to go back and update those docs for the images required for Disconnected Install?

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL. Updated the images required for Disconnected Install for 3.6.

**Comment by bmcelvee** [comment]
LGTM 🌟

**Comment by mburke5678** [comment]
[rev_history]
|xref:install_config/install/disconnected_install.adoc#install-config-install-disconnected-install[Disconnected Installation]
|Updated the list of required and optional images to download.
%

**Comment by vikram-redhat** [comment]
@mburke5678 - there was a typo in the rev history. I have fixed it for now.

**Comment by mburke5678** [comment]
See also https://github.com/openshift/openshift-docs/pull/5272

**Pull request merged**